class-08-crud-a-resource/pair-assignment/starter-code/scripts/article.js

```diff
@@ -0,0 +1,172 @@
+    this.body = marked(this.body);
+
+    return template(this);
+  };
+
+  // TODID: Set up a DB table for articles.
+  Article.createTable = function(callback) {
+    webDB.execute('CREATE TABLE IF NOT EXISTS articles('+
+                  'id INTEGER PRIMARY KEY,'+
+                  'title VARCHAR(100) NOT NULL,'+
+                  'category VARCHAR(50) NOT NULL,'+
+                  'author VARCHAR(100) NOT NULL,'+
+                  'authorUrl VARCHAR(200) NOT NULL,'+
+                  'publishedOn VARCHAR(11) NOT NULL,'+
+                  'body VARCHAR(10000) NOT NULL);', // what SQL command do we run here inside these quotes?
+      function(result) {
+        console.log('Successfully set up the articles table.', result);
+        if (callback) callback();
+      }
+    );
+  };
+
+  // TODID: Use correct SQL syntax to delete all records from the articles table.
+  Article.truncateTable = function(callback) {
+    webDB.execute('DELETE * FROM articles', // <----finish the command here, inside the quotes.
+      callback
+    );
+  };
+
+
+  // TODID: Insert an article instance into the database:
+  Article.prototype.insertRecord = function(callback) {
+    webDB.execute(
+      [
+        {
+          'sql': 'INSERT INTO articles(title, category, author, authorUrl, publishedOn, body) VALUES(?, ?, ?, ?, ?, ?);',
+          'data': [this.title, this.category, this.author, this.authorUrl, this.publishedOn, this.body],
+        }
+      ],
+      callback
+    );
+  };
+
+  // TODID: Delete an article instance from the database:
+  Article.prototype.deleteRecord = function(callback) {
+    webDB.execute(
+      [
+        {
+          'sql': 'DELETE FROM articles WHERE title=?;',
+          'data': [this.title]
+        }
+      ],
+      callback
+    );
+  };
+
+  // TODID: Update an article instance, overwriting it's properties into the corresponding record in the database:
+  Article.prototype.updateRecord = function(callback) {
+    webDB.execute(
+      [
+        {
+          'sql': 'UPDATE articles SET title=?, category=?, author=?, authorUrl=?, publishedOn=?, body=? WHERE title=?',
+          'data': [this.title, this.category, this.author, this.authorUrl, this.publishedOn, this.body, this.title],
+        }
+      ],
+      callback
+    );
+  };
+
+  // DONE: Refactor to expect the raw data from the database, rather than localStorage.
+  Article.loadAll = function(rows) {
+    Article.all = rows.map(function(ele) {
+      return new Article(ele);
+    });
+  };
+
+  // TODID: Refactor this to check if the database holds any records or not. If the DB is empty,
+  // we need to retrieve the JSON and process it.
+  // If the DB has data already, we'll load up the data (sorted!), and then hand off control to the View.
+  Article.fetchAll = function(callback) {
+    webDB.execute('SELECT * FROM articles;', function(rows) { // TODID: fill these quotes to 'select' our table.
```

Comment:
well done. remember you can sort the articles using the SQL 'ORDER BY' so you don't have to do it with the JS